src/modules/endpoint/models/CategoryModel.js

```diff
@@ -0,0 +1,65 @@
+class CategoryModel {
+  constructor ({ id, url = '', title = '', parent = 0, content = '', thumbnail = null, order = 0, availableLanguages = {} }) {
+    this._id = id
+    this._url = url
+    this._title = title
+    this._content = content
+    this._parent = parent
+    this._thumbnail = thumbnail
+    this._order = order
+    this._children = []
+    this._availableLanguages = availableLanguages
+  }
+
+  get thumbnail () {
+    return this._thumbnail
+  }
+
+  addChild (id) {
+    this._children.push(id)
+    this._children = this._children.sort((category1, category2) => category1.order - category2.order)
+  }
+
+  get id () {
+    return this._id
+  }
+
+  get url () {
+    return this._url
+  }
+
+  get title () {
+    return this._title
+  }
+
+  get content () {
+    return this._content
+  }
+
+  get parent () {
+    return this._parent
+  }
+
+  get children () {
+    return this._children
+  }
+
+  get order () {
+    return this._order
+  }
+
+  get availableLanguages () {
+    return this._availableLanguages
+  }
+
+  static getCategoryByPath (categories, path = '') {
+    return categories.find(category => category.url === encodeURI(path).toLowerCase())
```

Comment:
We rather need some url-normalizing here: Calling `/augsburg/de/` and `/augsburg/de` should both work

Reply:
Yeah we need to normalize here. Also encodeURI implementation differs from enviroment to environment. So it produces different outputs. As long as we don't have any user input we do not need to encode/decode our URIs

Reply:
I'm find with  getCategoryByPath. The method does not need to be named after the implementation.

Reply:
we need encodeURI because our cms provides as url for arabic and russian charachters UTF-8 encoded strings while our router gives us decoded url, i.e. "%d8%a7%d9%84%d8%a7%d8%b3%d8%aa%d9%82%d8%a8%d8%a7%d9%84-%d8%a7%d9%84%d8%a3%d9%88%d9%84%d9%8a" vs. الاستقبال-الأولي